src/test/python/jointGenerator/rockInput.py

```diff
@@ -0,0 +1,122 @@
+masterJoints = np.empty(0, dtype = object)
+joints = np.empty(0, dtype = object)
+
+# Generate joints that represent the bounds for the rock volume
+for face in facesData:
+    center = np.array([face[2], face[3], face[4]])
+    faces = np.append(faces, Joint(origin, center, face[0], face[1]))
+
+# Generate "master joint set". These will be copied based on joint spacing
+for joint in jointsData:
+    # localOrigin = np.array([boundingBox[0], boundingBox[1], boundingBox[2]])
+    masterJoints = np.append(masterJoints, 
+                             Joint(origin, origin, joint[0], joint[1], joint[2]))
+
+# Create complete joint set from master joints
+for joint in masterJoints:
+    cutoff = origin
+    count = 1.0
+    while (((cutoff[0] < boundingBox[3]) and (cutoff[0] > boundingBox[0])) and
+           ((cutoff[1] < boundingBox[4]) and (cutoff[1] > boundingBox[1])) and
+           ((cutoff[2] < boundingBox[5]) and (cutoff[2] > boundingBox[2]))):
+        # Increment center location based on joint spacing
+        centerIncrement = count * joint.spacing * joint.normalVec
+        count = count + 1.0
+        joints = np.append(joints,
+                           Joint(joint.localOrigin,
+                                 np.array([joint.center[0] + centerIncrement[0], 
+                                           joint.center[1] + centerIncrement[1],
+                                           joint.center[2] + centerIncrement[2]]),
+                                 joint.strike, joint.dip, joint.spacing))
+        cutoff = np.array([joint.center[0] + centerIncrement[0], 
+                           joint.center[1] + centerIncrement[1],
+                           joint.center[2] + centerIncrement[2]])
+
+    cutoff = origin
+    count = 1.0
+    while (((cutoff[0] < boundingBox[3]) and (cutoff[0] > boundingBox[0])) and
+           ((cutoff[1] < boundingBox[4]) and (cutoff[1] > boundingBox[1])) and
+           ((cutoff[2] < boundingBox[5]) and (cutoff[2] > boundingBox[2]))):
+        # Increment center location based on joint spacing
+        centerIncrement = count * joint.spacing * joint.normalVec
+        count = count + 1.0
+        joints = np.append(joints,
+                           Joint(joint.localOrigin,
+                                 np.array([joint.center[0] - centerIncrement[0], 
+                                           joint.center[1] - centerIncrement[1],
+                                           joint.center[2] - centerIncrement[2]]),
+                                 joint.strike, joint.dip, joint.spacing))
+        cutoff = np.array([joint.center[0] - centerIncrement[0], 
+                           joint.center[1] - centerIncrement[1],
+                           joint.center[2] - centerIncrement[2]])
+
+# Write face data to output file:
+f = open(outputFile, 'w')
+
+for face in faces:
+    # f.write("%.12f %.12f %.12f %.12f %.12f %.12f\n" % (face.normalVec[0], face.normalVec[1], face.normalVec[2], 
```

Comment:
Yeah, I'm still debugging a bit here. The issue was with reading in the inputs from a text file generated by this. I think I'm pretty close to figuring out what's going on here; looks like the numbers are being red in correctly. The issue is with the sign on the normal vectors and the rotations, but still need to get to the bottom of this.